File: go/strategy/Commuter.go
package main

import "fmt"

type CarStrategy struct{}

func (c CarStrategy) Transport() {
	fmt.Println("Transporting by car")
}

type BikeStrategy struct{}

func (b BikeStrategy) Transport() {
	fmt.Println("Transporting by bike")
}

type BusStrategy struct{}

func (b BusStrategy) Transport() {
	fmt.Println("Transporting by bus")
}

type Commuter struct {
	strategy TransportStrategy
}

func (c *Commuter) SetStrategy(strategy TransportStrategy) {
	c.strategy = strategy
}

func (c *Commuter) Commute() {
	if c.strategy == nil {
		fmt.Println("No transport strategy set")
		return
	}
	c.strategy.Transport()
}

File: python/strategy/Commuter.py
from TransportStrategy import TransportStrategy

class CarStrategy(TransportStrategy):
    def transport(self) -> None:
        print("Transporting by car")


class BikeStrategy(TransportStrategy):
    def transport(self) -> None:
        print("Transporting by bike")


class BusStrategy(TransportStrategy):
    def transport(self) -> None:
        print("Transporting by bus")
        
from typing import Optional


class Commuter:
    def __init__(self):
        self._strategy: Optional[TransportStrategy] = None

    def set_strategy(self, strategy: TransportStrategy) -> None:
        self._strategy = strategy

    def commute(self) -> None:
        if not self._strategy:
            print("No transport strategy set")
            return
        self._strategy.transport()

